aliyun-ecs/src/test/java/org/jclouds/aliyun/ecs/compute/functions/ImageInRegionToImageTest.java

```diff
@@ -0,0 +1,102 @@
+import org.jclouds.domain.LocationBuilder;
+import org.jclouds.domain.LocationScope;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.xml.bind.DatatypeConverter;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import static org.jclouds.aliyun.ecs.domain.Image.Status.AVAILABLE;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+@Test(groups = "unit", testName = "ImageInRegionToImageTest")
+public class ImageInRegionToImageTest {
+
+   private final Location region = new LocationBuilder()
+           .id(Regions.EU_CENTRAL_1.getName())
+           .description(Regions.EU_CENTRAL_1.getDescription())
+           .scope(LocationScope.REGION).build();
+   private final Supplier<Set<? extends Location>> locations = Suppliers.<Set<? extends Location>> ofInstance(ImmutableSet.of(region));
+
+   private ImageInRegionToImage imageInRegionToImage;
+
+   @BeforeMethod
+   public void setUp() {
+      imageInRegionToImage = new ImageInRegionToImage(locations);
+   }
+
+   @Test
+   public void testOsImageToImage() {
+      final org.jclouds.aliyun.ecs.domain.Image ecsImage = org.jclouds.aliyun.ecs.domain.Image.builder()
+              .id("centos_6_09_64_20G_alibase_20180326.vhd")
+              .description("")
+              .productCode("")
+              .osType("linux")
+              .architecture("x86_64")
+              .osName("CentOS  6.9 64位")
+              .imageOwnerAlias("system")
+              .progress("100%")
+              .isSupportCloudinit(true)
+              .usage("instance")
+              .creationTime(parseDate("2018-05-10T12:40:55Z"))
+              .imageVersion("")
+              .status(AVAILABLE)
+              .name("centos_6_09_64_20G_alibase_20180326.vhd")
+              .isSupportIoOptimizeds(true)
+              .isCopied(false)
+              .isSubscribed(false)
+              .isSelfShared(false)
+              .platform("CentOS")
+              .size(20)
+              .diskDeviceMappings(ImmutableMap.<String, List<DiskDeviceMapping>>of())
+              .tags(ImmutableMap.<String, List<Tag>>of())
+              .build();
+
+      final Image image = imageInRegionToImage.apply(ImageInRegion.create(Regions.EU_CENTRAL_1.getName(), ecsImage));
+      assertEquals(ecsImage.id(), image.getProviderId());
+      assertEquals(ecsImage.name(), image.getName());
+      assertEquals(Image.Status.AVAILABLE, image.getStatus());
+      final org.jclouds.compute.domain.OperatingSystem operatingSystem = image.getOperatingSystem();
+
+      assertEquals(ecsImage.osName(), operatingSystem.getName());
+      assertEquals(ecsImage.description(), operatingSystem.getDescription());
+      assertTrue(operatingSystem.is64Bit());
+      assertEquals(region, image.getLocation());
+   }
+
+   Date parseDate(final String dateString) {
+      return DatatypeConverter.parseDateTime(dateString).getTime();
+   }
```

Comment:
Add tests that check the special cases such as the `OTHER_OS_MAP`.

Reply:
done